File: C++/Test3.cpp
#include <iostream>

int main() 
{
    bool output = true;
    bool fortyTwo = false; 
    int lifes = 3, number;
    while (std::cin >> number)
    {
        if (output)
            std::cout << number << '\n';
        else
            break;
        if (number != 42)
            fortyTwo = true;
        else
        {
            if (fortyTwo) 
            {
                fortyTwo = false;
                lifes--;
                if (lifes == 0) 
                    output = false;
            }
        }
    }
    return 0;
}

File: Python/Test_3.py
output = True 
forty_two = False 
lifes = 3 
while True:
    try: 
        number = int(input())
        if output:
            print(number)
        if number != 42:
            forty_two = True
        else:
            if forty_two:
                forty_two = False
                lifes -= 1
                if lifes == 0:
                    output = False
    except:
        break
